src/app/loadout-builder/LoadoutBuilder.tsx

```diff
@@ -109,16 +106,39 @@ export default memo(function LoadoutBuilder({
   // TODO: if we're editing a loadout, grey out incompatible classes?
 
   // TODO: bundle these together into an LO context?
+  const loadoutParameters = loadout.parameters!;
   const modHashes = loadoutParameters.mods ?? emptyArray();
   const lockedExoticHash = loadoutParameters.exoticArmorHash;
+  const statConstraints =
+    loadoutParameters.statConstraints ?? defaultLoadoutParameters.statConstraints!;
   const autoStatMods = loadoutParameters.autoStatMods ?? false;
+  const statOrder = useMemo(() => statOrderFromStatConstraints(statConstraints), [statConstraints]);
 
   const selectedStore = stores.find((store) => store.id === selectedStoreId)!;
   const classType = selectedStore.classType;
   const loadouts = useRelevantLoadouts(selectedStore);
 
   const resolvedMods = useResolvedMods(defs, modHashes, selectedStoreId);
 
+  const itemCreationContext = useSelector(createItemContextSelector);
+  const allItems = useSelector(allItemsSelector);
+  const modsByBucket: {
+    [bucketHash: number]: number[];
+  } = loadoutParameters.modsByBucket ?? emptyObject();
+
+  // Turn loadout items into real DimItems, filtering out unequippable items
+  const [_items, subclass, _warnitems] = useMemo(
+    () =>
+      getItemsAndSubclassFromLoadout(
```

Comment:
This could probably be more efficient (since I only need the subclass!) but I'm not optimizing it for now in case I end up needing resolved items elsewhere.